src/evo/creditpool.cpp

```diff
@@ -249,14 +249,12 @@ bool CCreditPoolDiff::SetTarget(const CTransaction& tx, const CAmount blockRewar
 
 bool CCreditPoolDiff::Lock(const CTransaction& tx, TxValidationState& state)
 {
-    CAssetLockPayload assetLockTx;
-    if (!GetTxPayload(tx, assetLockTx)) {
+    if (CAssetLockPayload assetLockTx; !GetTxPayload(tx, assetLockTx)) {
```

Comment:
note for further refactoring: better to make GetTxPayload() to return std::expected or std::optional